site/en/start/go.md

```diff
@@ -0,0 +1,343 @@
+
+```bazel
+load("@rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "fortune",
+    srcs = ["fortune.go"],
+    importpath = "github.com/bazelbuild/examples/go-tutorial/stage3/fortune",
+    visibility = ["//visibility:public"],
+)
+
+go_test(
+    name = "fortune_test",
+    srcs = ["fortune_test.go"],
+    embed = [":fortune"],
+)
+```
+
+We have a new `fortune_test` target that uses the `go_test` rule to compile and link a test executable. `go_test` needs to compile `fortune.go` and `fortune_test.go` together with the same command, so we use the `embed` attribute here to incorporate the attributes of the `fortune` target into `fortune_test`. `embed` is most commonly used with `go_test` and `go_binary`, but it also works with `go_library`, which is sometimes useful for generated code.
+
+You may be wondering if the `embed` attribute is related to Go's [`embed`](https://pkg.go.dev/embed) package, which is used to access data files copied into an executable. This is an unfortunate name collision: rules_go's `embed` attribute was introduced before Go's `embed` package. Instead, rules_go uses the `embedsrcs` to list files that can be loaded with the `embed` package.
+
+Let's try running our test with `bazel test`:
+
+```posix-shell
+$ bazel test //fortune:fortune_test
+INFO: Analyzed target //fortune:fortune_test (0 packages loaded, 0 targets configured).
+INFO: Found 1 test target...
+Target //fortune:fortune_test up-to-date:
+  bazel-bin/fortune/fortune_test_/fortune_test
+INFO: Elapsed time: 0.168s, Critical Path: 0.00s
+INFO: 1 process: 1 internal.
+INFO: Build completed successfully, 1 total action
+//fortune:fortune_test                                          PASSED in 0.3s
+
+Executed 0 out of 1 test: 1 test passes.
+There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these are.
+```
+
+You can use the `...` wildcard to run all tests. Bazel will also build targets that aren't tests, so this can catch compile errors even in packages that don't have tests.
+
+```posix-shell
+$ bazel test //...
+```
+
+## Conclusion and further reading
+
+In this tutorial, we built and tested a small Go project with Bazel, and we learned some core Bazel concepts along the way.
+
+- To get started building other applications with Bazel, see the tutorials for [C++](/start/cpp), [Java](/start/java), [Android](/start/android-app), and [iOS](/start/ios-app).
+- You can also check the list of [recommended rules](/rules) for other languages.
+- For more information on Go, see the [rules_go](https://github.com/bazelbuild/rules_go) module, especially the [Core Go rules](https://github.com/bazelbuild/rules_go/blob/master/docs/go/core/rules.md) docuemntation.
+- To learn more about working with modules outside your project, see [external dependencies](/docs/external).
```

Comment:
Could make sense to say `Bazel modules` here as Go modules would be fetched with Gazelle.

Could you also mention external deps and reference the rules_go `bzlmod.md` guide for that?

Reply:
Thanks, made both changes.